File: LibraryManagementSystem_Java/LibraryManagementSystem/src/main/java/assignment/librarymanagementsystem/Book.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package assignment.librarymanagementsystem;

/**
 *
 * @author moonm
 */
class Book extends Item 
{
    private String title;
    private String author;

    public Book(String id, String title, String author, int pageCount, int totalCopies) {
        super(id, pageCount, totalCopies);
        this.title = title;
        this.author = author;
    }

    public String getTitle() { return title; }
    public String getAuthor() { return author; }
    public void setTitle(String title) { this.title = title; }
    public void setAuthor(String author) { this.author = author; }
}

File: LibraryManagementSystem-Python/models/book.py
from .item import Item

class Book(Item):
    def __init__(self, id: str, title: str, author: str, page_count: int = 0, total_copies: int = 1):
        super().__init__(id, page_count, total_copies)
        self.title = title
        self.author = author

    def get_title(self) -> str:
        return self.title

    def get_author(self) -> str:
        return self.author

    def set_title(self, title: str):
        self.title = title

    def set_author(self, author: str):
        self.author = author

    def __str__(self) -> str:
        return f"{self.title} by {self.author} (ID: {self.id})"

    def to_dict(self) -> dict:
        return {
            'id': self.id,
            'title': self.title,
            'author': self.author,
            'page_count': self.page_count,
            'total_copies': self.total_copies,
            'available_copies': self.available_copies
        }
